File: Playground/Play.py
import math
import time

# Function to check if a number is prime
def is_prime(num):
    if num <= 1:
        return False
    if num == 2:
        return True
    if num % 2 == 0:
        return False
    for i in range(3, int(math.sqrt(num)) + 1, 2):
        if num % i == 0:
            return False
    return True

# Function to calculate prime numbers up to a given limit
def calculate_primes(limit):
    primes = []
    for i in range(1, limit + 1):
        if is_prime(i):
            primes.append(i)
    return primes

# Function to run a time-consuming task
def perform_heavy_computation(iterations, prime_limit):
    for i in range(iterations):
        # Calculate primes up to prime_limit
        primes = calculate_primes(prime_limit)
        # Optional: Print progress to see the script is running (commented out to avoid slowing down the script further)
        # print(f"Iteration: {i + 1} - Number of primes found: {len(primes)}")

# Main function to start the performance test
def main():
    iterations = 10000       # Number of iterations to run
    prime_limit = 10000      # Limit up to which primes are calculated

    # Print start time
    print("Starting performance test...")

    # Record the start time
    start_time = time.time()

    # Run the heavy computation task
    perform_heavy_computation(iterations, prime_limit)

    # Record the end time
    end_time = time.time()

    # Print end time and duration
    print("Performance test completed.")
    print(f"Duration: {end_time - start_time:.2f} seconds")

# Run the main function
if __name__ == "__main__":
    main()

File: Playground/Play.lua
print("Playground initiation.");
-- Function to check if a number is prime
function is_prime(num)
  if num <= 1 then
      return false
  end
  if num == 2 then
      return true
  end
  if num % 2 == 0 then
      return false
  end
  for i = 3, math.sqrt(num), 2 do
      if num % i == 0 then
          return false
      end
  end
  return true
end

-- Function to calculate prime numbers up to a given limit
function calculate_primes(limit)
  local primes = {}
  for i = 1, limit do
      if is_prime(i) then
          table.insert(primes, i)
      end
  end
  return primes
end

-- Function to run a time-consuming task
function perform_heavy_computation(iterations, prime_limit)
  for i = 1, iterations do
      -- Calculate primes up to prime_limit
      local primes = calculate_primes(prime_limit)
      -- Optional: Print progress to see the script is running (commented out to avoid slowing down the script further)
      -- print("Iteration: " .. i .. " - Number of primes found: " .. #primes)
  end
end

-- Main function to start the performance test
function main()
  local iterations = 10000       -- Number of iterations to run
  local prime_limit = 10000     -- Limit up to which primes are calculated

  -- Print start time
  print("Starting performance test...")

  -- Run the heavy computation task
  perform_heavy_computation(iterations, prime_limit)

  -- Print end time
  print("Performance test completed.")
end

-- Run the main function
main()
